build_resolvers/lib/builder.dart

```diff
@@ -0,0 +1,72 @@
+// Copyright (c) 2023, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'dart:async';
+
+import 'package:build/build.dart';
+import 'package:convert/convert.dart';
+import 'package:crypto/crypto.dart';
+
+import 'src/build_asset_uri_resolver.dart';
+
+Builder transitiveDigestsBuilder(_) => _TransitiveDigestsBuilder();
+
+/// Computes a digest comprised of the current libraries digest as well as its
+/// transitive dependency digests, and writes it to a file next to the library.
+///
+/// For any dependency that has a transitive digest already written, we just use
+/// that and don't crawl its transitive deps, as the transitive digest includes
+/// all the information we need.
+class _TransitiveDigestsBuilder extends Builder {
+  @override
+  Future<void> build(BuildStep buildStep) async {
+    final seen = <AssetId>{buildStep.inputId};
+    final queue = [...seen];
+    final digestSink = AccumulatorSink<Digest>();
+    final byteSink = md5.startChunkedConversion(digestSink);
+
+    /// This gives us access to the resolvers cache of library dependencies,
+    /// which helps us to avoid duplicate parsing of libraries.
```

Comment:
[nit] 
```suggestion
    // This gives us access to the resolvers cache of library dependencies,
    // which helps us to avoid duplicate parsing of libraries.
```